woocommerce/woocommerce-ios
# [Mobile Payments] Handle plugin activate on IPP onboarding

Closes: #10566 

## Description

This PR updates the IPP onboarding `.pluginNotActivated` state by changing the current `Refresh After Activating` button for an `Activate WooCommerce Payments` button.

When the merchant taps this button, it will perform an API request to activate WooCommerce Payments, then it will refresh the onboarding state once completed and show the next steps (generally, "Finish Setup in Store Admin", since the plugin will be installed, and active, but not setup).

![Simulator Screen Recording - iPhone 11 Pro - 2023-09-05 at 11 29 30](https://github.com/woocommerce/woocommerce-ios/assets/3812076/0c93c248-72c0-4513-ba38-e50bd7182816)

## Caveats
Currently we have 2 plugins that can land on the "Plugin not activated" case: WCPay, and Stripe. We only cover WCPay since [we're removing the Stripe case on a different issue](https://github.com/woocommerce/woocommerce-ios/issues/10614). Only WCPay will offer the option to be activated after installing.

The way we handle this right now it's by returning the generic error if the Stripe view is shown and we attempt to activate it (since we can only do this for WCPay). The next step will be to remove the logic to make the Stripe case not an option. 

## Testing instructions
On a US store, with no WCPay or Stripe plugins installed (you can use https://atomicippwcpaytests.wpcomstaging.com/:

1. Go to Menu > Payments > See the notice at the bottom that says that In-Person Payments setup is incomplete. Continue setup. Tap the link.
2. See the `Install WooCommerce Payments` button. Tap
3. See the loading screen
4. See the `Activate WooCommerce Payments` button. Tap.
5. See the loading screen
6. See the `Finish Setup in Store Admin` button.
7. Confirm in the site's wp-admin that the WCPay plugin is installed and active

## Screenshots
| Install step | Activate step | Setup step |
|--------|--------|--------|
| <img width="299" alt="Screenshot 2023-09-05 at 11 40 04" src="https://github.com/woocommerce/woocommerce-ios/assets/3812076/dadf5a07-4200-4e62-b1f6-58be9e236a8f"> | <img width="296" alt="Screenshot 2023-09-05 at 11 40 18" src="https://github.com/woocommerce/woocommerce-ios/assets/3812076/2824deaa-8d64-4401-9852-22f19c186886"> | <img width="296" alt="Screenshot 2023-09-05 at 11 40 32" src="https://github.com/woocommerce/woocommerce-ios/assets/3812076/c8565cad-1f52-4eb3-84a4-a71c5c8eddc7"> |
